### PR TITLE
E2E tests: Add e2e jetpack connect flow

### PR DESCRIPTION
**Pull request by brbrr**
To be able to use/test Gutenpack blocks - we need to have a Jetpack connected site with a paid plan. This PR adds a connection & purchase flow to solve this configuration process.   

#### Changes proposed in this Pull Request:

* Adds a bunch of stuff!
* Adds a whole Jetpack connection flow with plan purchase

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?
* p9dueE-Ju-p2

#### Testing instructions:

* TBD
* Check travis E2E job
* decrypt config. Check the README file for the instructions. Are they hard to follow? 
* Make sure you have a running docker WP install with an activated jetpack (but not connected).
* run `yarn test-e2e` locally. use `PUPPETEER_HEADLESS=false` env variable to see the tests in action. Maybe check docs how to set up the tests

#### Proposed changelog entry for your changes:

* E2E: add Jetpack connect flow

### CODE REVIEW

**Comment by simison** [comment]
Should the bits that drive Calypso parts live in Calypso and be pulled in here as an NPM package? Will those connection bits run in both repositories?

**Comment by brbrr** [comment]
> Should the bits that drive Calypso parts live in Calypso and be pulled in here as an NPM package?

Maybe, I don't know :) At this point, we don't have any puppeteer tests in Calypso, so moving these bits there might complicate things and might provide no benefit. Later on, if we decide to run these tests in Calypso/WPCOM we definitely can benefit from sharing code. 

> Will those connection bits run in both repositories?

Ideally - yes, we would like to run these tests in _all_ environments, although I think we should not overcomplicate and overengineer at this point (there are no _real_ tests written yet). I wonder what @bsessions85 / @Stojdza think about it.

**Comment by bsessions85** [comment]
> Ideally - yes, we would like to run these tests in all environments, although I think we should not overcomplicate and overengineer at this point (there are no real tests written yet). I wonder what @bsessions85 / @Stojdza think about it.

@brbrr, Yes, I agree. It would be nice to share some code but just get it working and we can come back and make it work for us later. Eventually I would like to review all the tests and figure out a way to share everything between us if it makes sense

**Comment by simison** [comment]
Seems like a good way to start moving bits and bobs to [`/packages`](https://github.com/Automattic/wp-calypso/tree/585737ae318a8a4bb803b8a4483199c59bd2895f/packages) from [`/test/e2e`](https://github.com/Automattic/wp-calypso/tree/585737ae318a8a4bb803b8a4483199c59bd2895f/test/e2e) in small parts. :-)

File: tests/e2e/lib/flows/jetpack-connect.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+	await ( await HomePage.visit( page ) ).setSandboxModeForPayments( cookie );
+	await ( await WPLoginPage.visit( page ) ).login();
+	await ( await DashboardPage.init( page ) ).setSandboxModeForPayments( cookie, '.eu.ngrok.io' );
+	await ( await Sidebar.init( page ) ).selectJetpack();
+
+	const jetpackPage = await JetpackPage.init( page );
+	if ( await jetpackPage.isConnected() ) {
+		await jetpackPage.openMyPlan();
+		if ( await jetpackPage.isPlan( plan ) ) {
+			// eslint-disable-next-line no-console
```
Comment (simison) [inline tests/e2e/lib/flows/jetpack-connect.js]: If there are likely going to be lots of these in future, you could also add a new override here:

https://github.com/Automattic/jetpack/blob/91939c13e6d278629deb444ff3538677deb009d8/.eslintrc.js#L102

```js
files: [ ' tests/e2e/**/*' ],
rules: {
	'no-console': 0
}
```

Comment (brbrr) [reply tests/e2e/lib/flows/jetpack-connect.js]: Thanks! I'd like to switch to some fancier logging solution, so for now, it's fine :)

Comment (simison) [reply tests/e2e/lib/flows/jetpack-connect.js]: Sounds great! :-)

**Comment by zinigor** [comment]
This looks good, tests seem to be able to run, but are you expecting the tests to fail on Travis currently? 

The reports have this:
```
FAIL tests/e2e/specs/dummy.test.js
  ● Test suite failed to run
    TypeError: (0 , _slack.registerScreenshotReporter) is not a function
```

**Comment by brbrr** [comment]
@zinigor oh, seems I screwed up the rebase. should be fixed in a19bd5c

**Review by zinigor — APPROVED** [review]
The suite seems to be running properly on Travis now, let's get it in, thanks!
